depends/thirdparty/orc/tools/src/HdfsFileMemory.cc

```diff
@@ -0,0 +1,151 @@
+  uint64_t maxMemory;
+
+public:
+  char* malloc(uint64_t size) override {
+    char* p = static_cast<char*>(std::malloc(size));
+    blocks[p] = size ;
+    totalMemory += size;
+    if (maxMemory < totalMemory) {
+      maxMemory = totalMemory;
+    }
+    return p;
+  }
+
+  void free(char* p) override {
+    std::free(p);
+    totalMemory -= blocks[p] ;
+    blocks.erase(p);
+  }
+
+  uint64_t getMaxMemory() {
+    return maxMemory ;
+  }
+
+  TestMemoryPool(): totalMemory(0), maxMemory(0) {}
+  ~TestMemoryPool();
+};
+
+TestMemoryPool::~TestMemoryPool() {}
+
+void processFile(const char* filename,
+                 const std::list<uint64_t>& cols,
+                 uint32_t batchSize) {
+  orc::ReaderOptions opts;
+  if (cols.size() > 0) {
+    opts.include(cols);
+  }
+  std::unique_ptr<orc::MemoryPool> pool(new TestMemoryPool());
+  opts.setMemoryPool(*(pool.get()));
+    
+  auto addr = std::getenv("LIBHDFS3_NAMENODE_ADDRESS");
+  std::string nn = "localhost";
+  if(addr != NULL) {
+    nn = std::string(addr);
+  }
+  auto portStr = std::getenv("LIBHDFS3_NAMENODE_PORT");
+  tPort port = 8020;
+  if (portStr != NULL) {
+    port = static_cast<tPort>(atoi(portStr));
+  }
+  hdfsFS fs = hdfsConnect(nn.c_str(), port);
+
+  std::unique_ptr<orc::Reader> reader =
+    orc::createReader(orc::readHdfsFile(fs, std::string(filename)), opts);
+
+  std::unique_ptr<orc::ColumnVectorBatch> batch =
+      reader->createRowBatch(batchSize);
+  uint64_t readerMemory = reader->getMemoryUse();
+  uint64_t batchMemory = batch->getMemoryUsage();
+  while (reader->next(*batch)) {}
+  uint64_t actualMemory =
+      static_cast<TestMemoryPool*>(pool.get())->getMaxMemory();
+  std::cout << "Reader memory estimate: " << readerMemory
+            << "\nBatch memory estimate:  " ;
+  if (batch->hasVariableLength()) {
+    std::cout << "Cannot estimate because reading ARRAY or MAP columns";
+  } else {
+    std::cout << batchMemory
+              << "\nTotal memory estimate:  " << readerMemory + batchMemory;
+  }
+  std::cout << "\nActual max memory used: " << actualMemory << "\n";
+}
+
+int main(int argc, char* argv[]) {
+  if (argc < 2) {
+    std::cout << "Usage: hdfs-orc-memory [--columns=column1,column2,...] "
+        << "[--batch=rows_in_batch] <filename> \n";
+    return 1;
+  }
+
+  const std::string COLUMNS_PREFIX = "--columns=";
+  const std::string BATCH_PREFIX = "--batch=";
+  char* filename = ORC_NULLPTR;
+
+  // Default parameters
+  std::list<uint64_t> cols;
+  uint32_t batchSize = 1000;
+
+  // Read command-line options
+  char *param, *value;
+  for (int i = 1; i < argc; i++) {
+    if ( (param = std::strstr(argv[i], COLUMNS_PREFIX.c_str())) ) {
+      value = std::strtok(param+COLUMNS_PREFIX.length(), "," );
+      while (value) {
+        cols.push_back(static_cast<uint64_t>(std::atoi(value)));
+        value = std::strtok(ORC_NULLPTR, "," );
+      }
+    } else if ( (param=strstr(argv[i], BATCH_PREFIX.c_str())) ) {
+      batchSize =
```

Comment:
@ictmalili It is already configurable now, assignment here using `BATCH_PREFIX = '--batch'`.